### PR TITLE
feat: create new TopicConfiguration class for configuring TopicClient

### PR DESCRIPTION
**Pull request by cprice404**
Prior to this commit, the TopicClient was using a CacheClient
Configuration object for configuration. There are many settings
on the CacheClient configuration that aren't relevant for Topics,
and soon the reverse will likely be true as well. This commit changes
the TopicClient constructor to take a new TopicConfiguration object
which will give us the ability to define the configuration API for the
two clients independently going forward.

### CODE REVIEW

**Review by malandis — APPROVED** [review]

File: packages/client-sdk-web/src/internal/pubsub-client.ts [hunk 1]
```diff
@@ -1,9 +1,7 @@
 import * as pubsub from '@gomomento/generated-types-webtext/dist/CachepubsubServiceClientPb';
```
Comment (eaddingtonwhite) [inline packages/client-sdk-web/src/internal/pubsub-client.ts:1]: not related to this pr but we should probably rename this client purge pubsub and just call this topic.

File: packages/client-sdk-nodejs/src/internal/pubsub-client.ts [hunk 2]
```diff
@@ -3,7 +3,6 @@ import grpcPubsub = pubsub.cache_client.pubsub;
 // older versions of node don't have the global util variables https://github.com/nodejs/node/issues/20365
```
Comment (eaddingtonwhite) [inline packages/client-sdk-nodejs/src/internal/pubsub-client.ts:3]: same thought here s/pubsub/topic/ can be another pr.

File: packages/client-sdk-web/src/internal/pubsub-client.ts (hunk 1 above)
Comment (cprice404) [reply packages/client-sdk-web/src/internal/pubsub-client.ts:1]: it's internal, so not public API, but sure we could do that

**Review by eaddingtonwhite — APPROVED** [review]
